src/main/scala/org/apache/spark/sql/kinesis/HDFSMetadataCommitter.scala

```diff
@@ -165,4 +211,35 @@ class HDFSMetadataCommitter[T <: AnyRef : ClassTag](path: String,
     }
   }
 
+  /** Helper method to retry with exponential backoff  */
+  def withRetry[ T ](message: String, ignoreException: Boolean = true)(body: => T): T = {
+    var retryCount = 0
+    var result: Option[ T ] = None
+    var waitTimeInterval = retryIntervalMs
+    var lastError: Throwable = null
+
+    def isMaxRetryDone = retryCount >= numRetries
+
+    while (result.isEmpty && !isMaxRetryDone) {
+      if ( retryCount > 0 ) { // wait only if this is a retry
+        Thread.sleep(waitTimeInterval)
+        waitTimeInterval = scala.math.min(waitTimeInterval * 2, maxRetryIntervalMs)
```

Comment:
should we make the exponential factor configurable?

Reply:
I think its okay for now.